src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -247,6 +247,7 @@ void DatasetManager::HandleSet(Coap::Header &aHeader, Message &aMessage, const I
     uint16_t offset = aMessage.GetOffset();
     Tlv::Type type;
     StateTlv::State state = StateTlv::kAccept;
+    Dataset local = mLocal;
```

Comment:
I think we could reduce stack pressure by applying each TLV directly to `mLocal`.  Of course, you should still do the timestamp check and any other checking before you go through that process.